server/src/org/apache/cloudstack/outofbandmanagement/OutOfBandManagementServiceImpl.java

```diff
@@ -0,0 +1,532 @@
+                    response.setResultDescription(driverResponse.getResult());
+                    response.setId(host.getUuid());
+                    return response;
+                } finally {
+                    outOfBandManagementHostLock.unlock();
+                }
+            } else {
+                LOG.error("Unable to acquire synchronization lock to change out-of-band management password for host id: " + host.getId());
+                throw new CloudRuntimeException(String.format("Unable to acquire lock to change out-of-band management password for host (%s), please try after some time.", host.getUuid()));
+            }
+        } finally {
+            outOfBandManagementHostLock.releaseRef();
+        }
+    }
+
+    @Override
+    public String getName() {
+        return name;
+    }
+
+    @Override
+    public long getId() {
+        return serviceId;
+    }
+
+    @Override
+    public boolean configure(final String name, final Map<String, Object> params) throws ConfigurationException {
+        this.name = name;
+        this.serviceId = ManagementServerNode.getManagementServerId();
+
+        final int poolSize = OutOfBandManagementSyncThreadPoolSize.value();
+
+        hostAlertCache = CacheBuilder.newBuilder()
+                .concurrencyLevel(4)
+                .weakKeys()
+                .maximumSize(100 * poolSize)
+                .expireAfterWrite(1, TimeUnit.DAYS)
+                .build();
+
+        backgroundSyncExecutor = new ThreadPoolExecutor(poolSize, poolSize,
+                0L, TimeUnit.MILLISECONDS,
+                new ArrayBlockingQueue<Runnable>(10 * poolSize, true), new ThreadPoolExecutor.CallerRunsPolicy());
+
+        LOG.info("Starting out-of-band management background sync executor with thread pool-size=" + poolSize + " and background sync thread interval=" + OutOfBandManagementSyncThreadInterval.value() + "s");
+        return true;
+    }
+
+    @Override
+    public boolean start() {
+        initializeDriversMap();
+        return true;
+    }
+
+    @Override
+    public boolean stop() {
+        backgroundSyncExecutor.shutdown();
+        outOfBandManagementDao.expireOutOfBandManagementOwnershipByServer(getId());
+        return true;
+    }
+
+    @Override
+    public String getConfigComponentName() {
+        return OutOfBandManagementServiceImpl.class.getSimpleName();
+    }
+
+    @Override
+    public ConfigKey<?>[] getConfigKeys() {
+        return new ConfigKey<?>[] {OutOfBandManagementActionTimeout, OutOfBandManagementSyncThreadInterval, OutOfBandManagementSyncThreadPoolSize};
+    }
+
+    public List<OutOfBandManagementDriver> getOutOfBandManagementDrivers() {
+        return outOfBandManagementDrivers;
+    }
+
+    public void setOutOfBandManagementDrivers(List<OutOfBandManagementDriver> outOfBandManagementDrivers) {
+        this.outOfBandManagementDrivers = outOfBandManagementDrivers;
```

Comment:
Make a defensive copy of ``outOfBandManagementDrivers`` to ensure changes made by the caller do not impact this object.